### ISSUE TITLE
how to deal with occlusion when nmm is applied

### ISSUE THREAD

**Issue by junghyun-avikus** [obss/sahi#560]
I guess when two objects are occluded (1 small object, 1 large object), nmm will merge those to objects into one bounding box. (In case where the smaller object lies on the other larger object). Are there any ideas or potential solutions to this issue?

**Comment by fcakyon** [comment]
@junghyun-avikus if the classes are same, yes this limitation applies. You can use NMS instead of NMM to handle these situations better: https://github.com/obss/sahi/blob/072ea62bc14646738ec95437e038f88690a0aa4c/sahi/predict.py#L163

**Comment by junghyun-avikus** [comment]
Yes, but the thing is that the way I am feeding input image to a network is 3 tiled (divided into slices) images and 1 full image. In this case, NMS won't get rid of a large object that presents across the entire frame. So I was trying to come up with an idea that can solve the issue, but I could't come up with one.

BTW, slicing works really well for detecting smaller objects and in general. I appreciate your great work.

**Comment by fcakyon** [comment]
@junghyun-avikus how about using NMS with IOS match metric: https://github.com/obss/sahi/blob/main/sahi/predict.py#L132

**Comment by junghyun-avikus** [comment]
@fcakyon Yes, but that creates another problem that objects in full frame might get removed when parts of an object in tilled frame have higher confidence score.
